ast-nodes/range.dart

```diff
@@ -9,13 +11,32 @@ class Range implements Node {
 
   Range(this.start, this.end);
 
-  // TODO: implement .parse()
+  factory Range.parse(Iterable<Token> tokens) {
+    var iterator = tokens.iterator;
+    if (!iterator.moveNext()) {
+      throw SyntaxError(iterator.current, "Expected '..' in range");
+    }
+
+    var expStart = Expression.parse(
+        consumeUntil(iterator, RegExp("\^" + RegExp.escape("..") + "\$")));
```

Comment:
This isn't going to work, we have all periods as separate tokens. You'll have to manually consume tokens here, remembering the previous one